src/main/java/edu/kis/powp/jobs2d/drivers/visitor/DriversCountVisitor.java

```diff
@@ -6,41 +6,27 @@
 import edu.kis.powp.jobs2d.drivers.logger.LoggerDriver;
 
 import java.util.List;
-import java.util.stream.Collectors;
 
 public class DriversCountVisitor implements IDriverVisitor {
-    private long count = 0;
-
-    public long getDriversCount() {
-        long tmpCount = count;
-        count = 0;
-        return tmpCount;
-    }
-
     @Override
-    public void visit(IDriverComposite driver) {
-        List<VisitableJob2dDriver> drivers = driver.getDrivers().stream()
-                .map(driver1 -> (VisitableJob2dDriver) driver1)
-                .collect(Collectors.toList());
-
-        drivers.forEach(driver1 -> driver1.accept(this));
+    public long visit(IDriverComposite driver) {
+        List<VisitableJob2dDriver> drivers = driver.getDrivers();
+        return 1 + drivers.stream().mapToLong(d -> d.accept(this)).sum();
     }
 
     @Override
-    public void visit(DriverDecorator driver) {
-        VisitableJob2dDriver wrappedDriver = (VisitableJob2dDriver) driver.getDriver();
-        wrappedDriver.accept(this);
-        count++;
+    public long visit(DriverDecorator driver) {
+        VisitableJob2dDriver wrappedDriver = driver.getDriver();
+        return 1 + wrappedDriver.accept(this);
     }
 
-
     @Override
-    public void visit(VisitableJob2dDriver driver) {
-        count++;
+    public long visit(VisitableJob2dDriver driver) {
+        return 1;
     }
 
     @Override
-    public void visit(LoggerDriver driver) {
-        count++;
+    public long visit(LoggerDriver driver) {
+        return 1;
```

Comment:
the approach is correct, however result should be read using getter (rather then be returned by accept)